src/vpddbenv.cpp

```diff
@@ -82,7 +82,9 @@ namespace lsvpd
 		}
 
 		for( unsigned int seconds = 1;; seconds++ ) {
-			rc = sqlite3_open( mDbPath.c_str( ), &mpVpdDb );
+			rc = sqlite3_open_v2( mDbPath.c_str( ), &mpVpdDb,
+					readOnly ? SQLITE_OPEN_READONLY : SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE,
+					NULL );
```

Comment:
The sqlite database is closed during destructor and I see it still uses sqlite3_close(), do we need to change that as well to sqlite_close_v2() ?

Reply:
The v2 in sqlite_cllose_v2 is different v2:

** Ideally, applications should [sqlite3_finalize | finalize] all
** [prepared statements], [sqlite3_blob_close | close] all [BLOB handles], and
** [sqlite3_backup_finish | finish] all [sqlite3_backup] objects associated
** with the [sqlite3] object prior to attempting to close the object.
** ^If the database connection is associated with unfinalized prepared
** statements, BLOB handlers, and/or unfinished sqlite3_backup objects then
** sqlite3_close() will leave the database connection open and return
** [SQLITE_BUSY]. ^If sqlite3_close_v2() is called with unfinalized prepared
** statements, unclosed BLOB handlers, and/or unfinished sqlite3_backups,
** it returns [SQLITE_OK] regardless, but instead of deallocating the database
** connection immediately, it marks the database connection as an unusable
** "zombie" and makes arrangements to automatically deallocate the database
** connection after all prepared statements are finalized, all BLOB handles
** are closed, and all backups have finished. The sqlite3_close_v2() interface
** is intended for use with host languages that are garbage collected, and
** where the order in which destructors are called is arbitrary.

It make no difference for out use case because if such objects that prevent closing down the connection exist there is no code for closing those.

Reply:
ok got it.